File: lectures/lect2/Prototype/src/SUV.java
public class SUV extends Car {

    private final boolean isAWD;

    public SUV(String mark, int power, int torque, boolean isAWD) {
        super(mark, power, torque);
        this.isAWD = isAWD;
    }

    public SUV(SUV other) {
        super(other);
        this.isAWD = other.isAWD;
    }

    @Override
    public Car clone() {
        return new SUV(this);
    }

    @Override
    public String toString() {
        return "SUV{" +
                super.toString() +
                "isAWD=" + isAWD +
                '}';
    }
}

File: Cpp/lab05_Prototype/task_2_1/src/suv.cpp
#include "suv.hpp"


SUV::SUV(std::string const& mark, int power, int torque, bool isAWD)
: Car(mark, power, torque), isAWD(isAWD) {}

SUV::SUV(SUV const& other)
: Car(other), isAWD(other.isAWD) {}

SUV* SUV::clone() const
{
  return new SUV(*this);
}

std::ostream& operator<<(std::ostream& os, SUV const& suv)
{
  os << "SUV{"
     << static_cast<Car const&>(suv)
     << ", isAWD=" << (suv.isAWD ? "true" : "false")
     << "}";

  return os;
}
